Automattic/studio
# Add Assistant chat interface

Resolves Automattic/dotcom-forge#7323

## Proposed Changes

- Updates ContentTabAssistant to display chat messages
- Adds `useAssistant` hook to store and retrieve chat messages (currently using localStorage, but expected to change)
- Introduces basic responsive design functionality 
- Updates general spacing of all tabs
- (For testing) Simulates a server response
- Moves icons to an icons directory


https://github.com/Automattic/studio/assets/643285/348f240f-92cd-4768-a8ac-e9ada1bdeecd



## Testing Instructions
Proposed requirements, from https://github.com/Automattic/dotcom-forge/issues/7323:

**When I click on AI Assistant tab, I can see the chat interface**
1. Start Studio and add site
2. Navigate to the Assistant tab and observe a text input with an empty chat window

**Pressing enter on the text input will send the message**
1. Start Studio and add site
2. Navigate to the Assistant tab and type a message
3. Hit enter and observe the message appears in the chat window
4. Observe a simulated server response

**The messages are stored locally**
1. In the left Sidebar, select a different site and navigate to that site's Assistant tab
2. Observe that the chat is empty (for a new site)
4. Start a new chat in a different site, or close the app and reopen it
5. Observe the chat is persisted (alternatively, use the Developer Tools to view the localStorage contents)

**When I change the selected site, I can see a new chat interface with the messages I sent related to that site**
1. In the left Sidebar, select a different site and navigate to that site's Assistant tab
2. Observe that the chat is empty (for a new site)
4. Start a new chat in a different site, or close the app and reopen it
5. Observe the chat is persisted (alternatively, use the Developer Tools to view the localStorage contents)

**The clear chat button will delete the messages for that site**
1. Create a chat conversation with several messages
2. Click the three-dots icon to the right of the text input
3. Observe that the chat is cleared from the window


## Pre-merge Checklist




## Todo
- [x] Check for any layout style regressions, including Windows and RTL layouts
- [ ] ~Add the user id as a key to the chat storage so that chats are unique to a user's account~ _Deferred to_ https://github.com/Automattic/dotcom-forge/issues/7532
- [x] Seek and incorporate design feedback on the current UX
